app/services/utils/money_with_precision.rb

```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+module Utils
+  class MoneyWithPrecision < Money
+    self.default_infinite_precision = true
```

Comment:
this will add the `infinite_precision` for the whole instance, on some value it will not match what we expect because we do not want the precise value but the rounded one

Reply:
In that case, `MoneyWithPrecision` class, the whole instance needs to be with precision ;)

Reply:
yep, only MoneyWithPrecision will have this property set. Money class will be untouched.